### DISCUSSION TITLE
Differences between CallExpr and MethodCallExpr in Javascript

### DISCUSSION THREAD

**Discussion by mrthankyou** [github/securitylab#228] [Q&A] [answered]
EDIT: I've discovered that I can look at the source code for MethodCallExpr and understand through the characteristic predicate what a method call expression means. Once I complete my research I'll post my answer below.

Hi,

I'm currently reviewing the docs and see that inside the CodeQL Javascript code there is both a `CallExpr` and a subtype `MethodCallExpr`. I know that `CallExpr` classes are used for direct calls and `MethodCallExpr` are used for indirect calls. 

```
// Below is the documentation I found on CallExpr and MethodCallExpr:

// A CallExpr has the following examples:
//    f()
//    require('express')()
//    x.f()

// A MethodCallExpr has the following examples:
//    Object.create(null)
//    [1, 2, 3].forEach(alert);

```

Here's my question: what's the difference between the two classes?

**Comment by criemen** [comment, marked as answer]
Hi mrthankyou,
I am sorry nobody answered here earlier. Please let me know if you still need help with this issue or not.
